File: backend/app/schemas/transfer.py
from datetime import datetime
from pydantic import BaseModel

class TransferCreate(BaseModel):
    send_country: str
    receive_country: str
    send_currency: str
    receive_currency: str
    send_amount: float
    recipient_name: str
    recipient_phone: str

class TransferOut(BaseModel):
    id: int
    user_id: int
    send_country: str
    receive_country: str
    send_currency: str
    receive_currency: str
    send_amount: float
    rate_used: float | None
    fee_used: float | None
    total_payable: float | None
    receive_amount: float | None
    recipient_name: str
    recipient_phone: str
    provider: str
    status: str
    fail_reason: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

class TransferStatusUpdate(BaseModel):
    status: str
    fail_reason: str | None = None

class TransferCreateFromQuote(BaseModel):
    quote_id: int

File: frontend/zuripay-frontend/src/types/transfer.ts
export type TransferStatus = "initiated" | "payment_pending" | "processing" | "sent" | "received" | "failed" | "cancelled";

export interface TransferCreate {
  send_country: string;
  receive_country: string;
  send_currency: string;
  receive_currency: string;
  send_amount: number;
  send_method?: string;
  receive_method?: string;
  recipient_name: string;
  recipient_phone: string;
}

export interface Transfer {
  id: number;
  user_id: number;
  status: TransferStatus;
  send_country: string;
  receive_country: string;
  send_currency: string;
  receive_currency: string;
  send_amount: number;
  receive_amount?: number;
  recipient_name: string;
  recipient_phone: string;
  provider: string;
  fx_rate: number;
  fee_amount: number;
  failure_reason?: string;
  created_at?: string;
  updated_at?: string;
}
